api/envoy/extensions/filters/http/composite/v3/composite.proto

```diff
@@ -29,8 +32,27 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message Composite {
 }
 
+// Configuration for an extension configuration discovery service with name.
+message DynamicConfig {
+  // The name of the extension configuration. It also serves as a resource name in ExtensionConfigDS.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Configuration source specifier for an extension configuration discovery
+  // service. In case of a failure and without the default configuration, the
+  // listener closes the connections.
+  config.core.v3.ExtensionConfigSource config_discovery = 2;
+}
+
 // Composite match action (see :ref:`matching docs <arch_overview_matching_api>` for more info on match actions).
 // This specifies the filter configuration of the filter that the composite filter should delegate filter interactions to.
 message ExecuteFilterAction {
-  config.core.v3.TypedExtensionConfig typed_config = 1;
+  // Filter specific configuration which depends on the filter being
+  // instantiated. See the supported filters for further documentation.
+  // [#extension-category: envoy.filters.listener,envoy.filters.udp_listener]
+  config.core.v3.TypedExtensionConfig typed_config = 1
+      [(udpa.annotations.field_migrate).oneof_promotion = "config_type"];
+
+  // Dynamic configuration of filter obtained via extension configuration discovery
+  // service.
```

Comment:
Forgot one small nit: please add a comment here and to the field above, that only one must be defined. Consider using the [oneof promotion](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md#field-level) attribute.
This should also be accompanied with a config-rejection test.

Reply:
added. config rejection was added already. PTAL.